CONTRIBUTING.md

```diff
@@ -147,6 +147,55 @@ enabled as a plugin:
 #![plugin(clippy)]
 ```
 
+### How Clippy works
+
+Clippy is a [rustc compiler plugin][compiler_plugin]. The main entry point is at [`src/lib.rs`][main_entry]. In there, the lint registration is delegated to the [`clippy_lints`][lint_crate] crate.
+
+[`clippy_lints/src/lib.rs`][lint_crate_entry] imports all the different lint modules and registers them with the rustc plugin registry. For example, the [`else_if_without_else`][else_if_without_else] lint is registered like this:
+
+```rust
+// ./clippy_lints/src/lib.rs
+
+// ...
+pub mod else_if_without_else;
+// ...
+
+pub fn register_plugins(reg: &mut rustc_plugin::Registry) {
+    // ...
+    reg.register_early_lint_pass(box else_if_without_else::ElseIfWithoutElse);
+    // ...
+
+    reg.register_lint_group("clippy_restriction", vec![
```

Comment:
Worth mentioning that this code is autogenerated

Reply:
Hm, I'm not sure I understand how that code would be autogenerated. It's a normal call to [`Registry.register_lint_group`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_plugin/registry/struct.Registry.html#method.register_lint_group), no?

Reply:
oh, *that* kind of autogenerated! I've always been modifying it manually until now.. Makes sense to add it :+1: